### ISSUE TITLE
dataset

### ISSUE THREAD

**Issue by 0000it** [tongshw/GeoDistill#2]
Thank you for sharing your excellent work and the accompanying code.  
I am very interested in reproducing your experiments and extending your research.  

I noticed that the dataset used in the paper/code is not included in this repository.  
Could you please let me know if the dataset can be shared, or if there is a way to access it?  
If it is private or requires permission, I would greatly appreciate any guidance on how I may obtain it.  

Thank you very much for your time and support.

**Comment by tongshw** [comment]
Thank you for your attention and sorry for the late reply.
VIGOR dataset can be found here:https://github.com/Jeff-Zilence/VIGOR
VIGOR Dataset Corrected Labels: https://github.com/tudelft-iv/SliceMatch.
KITTI dataset: https://github.com/YujiaoShi/HighlyAccurate

**Comment by 0000it** [comment]
Hello, I replicated Kitti's g2sweakly according to the code you provided. However, I found that the loss oscillation was very severe. Besides, the mean error and median error in the first epoch were actually the best. But as the training continued, the loss decreased while the error increased. I haven't modified the code. Is this normal?

**Comment by 0000it** [comment]
<img width="1037" height="720" alt="Image" src="https://github.com/user-attachments/assets/f63bfc70-eddd-4825-b182-09689892d78f" />

**Comment by tongshw** [comment]
This phenomenon is normal if the base model is g2sweakly. The data in our paper also shows that our method doesn't significantly improve performance on g2sweakly+KITTI because the initial performance of the base model is too poor, and the supervision signal has too much noise. If you change to a base model with better initial performance, you'll see a gradual improvement in performance.

**Comment by 0000it** [comment]
<img width="1572" height="859" alt="Image" src="https://github.com/user-attachments/assets/11898622-d041-4844-9716-40588773f5c2" />
Hello, sorry to bother you. We used the KITTI dataset and trained the model with GeoDistill, but after 5 epochs, the error is still around 19, which is not as good as the 11–12 reported in the paper. May I ask how many epochs you trained for? I followed your provided code exactly without any modifications, and ran the training using the command:
`python -u train_kitti.py --train True --cross_area`

Perhaps you know where the problem lies with us

**Comment by tongshw** [comment]
> <img alt="Image" width="1572" height="859" src="https://private-user-images.githubusercontent.com/199059088/510717469-11898622-d041-4844-9716-40588773f5c2.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI0MzU1OTksIm5iZiI6MTc2MjQzNTI5OSwicGF0aCI6Ii8xOTkwNTkwODgvNTEwNzE3NDY5LTExODk4NjIyLWQwNDEtNDg0NC05NzE2LTQwNTg4NzczZjVjMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEwNlQxMzIxMzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NWY5ZjlhOWQ2YjgzY2ViMjJhZWYxNDRiZjM3NWI0ZDAyZjNhMDdmMWI2NTg4ODliM2E3NmM5MmZkMmI3OGRhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.sdDZuDFedFSFY5sKQ8UOwlF7nHDBxS37t0M-CTL-aNU"> Hello, sorry to bother you. We used the KITTI dataset and trained the model with GeoDistill, but after 5 epochs, the error is still around 19, which is not as good as the 11–12 reported in the paper. May I ask how many epochs you trained for? I followed your provided code exactly without any modifications, and ran the training using the command: `python -u train_kitti.py --train True --cross_area`
> Perhaps you know where the problem lies with us

What pth are you using in your base model? Perhaps you should first check if inference using the base model's weights performs correctly, and then use GeoDistill to retrain that base model.

**Comment by 0000it** [comment]
I'm using the GeoDistill base model to train DINO. The model inferences correctly, but it can't train to achieve the desired performance. Do you mean the code needs to be changed? Did you freeze the parameters of the base model? And then I need to undo this and retrain dinov2_vitl14_pretrain? Is that what you mean?

**Comment by tongshw** [comment]
> I'm using the GeoDistill base model to train DINO. The model inferences correctly, but it can't train to achieve the desired performance. Do you mean the code needs to be changed? Did you freeze the parameters of the base model? And then I need to undo this and retrain dinov2_vitl14_pretrain? Is that what you mean?

The base model is G2SWeakly or CCVPE. 19 is like a random guess, I think perhaps the initial weights for teacher/student were not loaded correctly.

**Comment by 0000it** [comment]
I apologize, blogger, but I think you misunderstood me. The basic model I used wasn't GSWeakly or CCVPE at all; I used `train_geodistill` from your code `train_kitti.py`, with GeoDistill (DINO) as the base model. Even following the code you provided exactly, I still couldn't reproduce the results shown in Table 1 of the paper. Could you please check the code in the paper? I followed the code you provided for training the Kitti dataset exactly. Initially, shouldn't `teacher` and `student` have no weights?

**Comment by 0000it** [comment]
> 您在基础模型中使用了什么 pth？也许您应该首先检查使用基础模型权重的推理是否正确执行，然后使用 GeoDistill 重新训练该基础模型。

I'm very sorry to disturb you again. When your paper mentioned training with G2SWeakly, "For the G2Sweakly-Dino variant, the pre-trained DINOv2-b14 was used as the feature extractor." During the training period, the weights of DINOv2 remain frozen and an additional DPT module is added to fine-tune the DINO features. Does this indicate that the model weights of DINO do not need to be retrained? Is it a problem given by your code? I'm completely unable to reproduce any experimental result from the KITTI dataset? Could it be that I didn't do something right?

**Comment by tongshw** [comment]
> I apologize, blogger, but I think you misunderstood me. The basic model I used wasn't GSWeakly or CCVPE at all; I used `train_geodistill` from your code `train_kitti.py`, with GeoDistill (DINO) as the base model. Even following the code you provided exactly, I still couldn't reproduce the results shown in Table 1 of the paper. Could you please check the code in the paper? I followed the code you provided for training the Kitti dataset exactly. Initially, shouldn't `teacher` and `student` have no weights?

The paper explicitly states that the weights for teacher and student are initialized from the same pre-trained base model(the base model is G2SWeakly in this repo, so you should load teacher/student's weight from a pretrained model before retrain with Geodistill), not randomly.

**Comment by 0000it** [comment]
> > I apologize, blogger, but I think you misunderstood me. The basic model I used wasn't GSWeakly or CCVPE at all; I used `train_geodistill` from your code `train_kitti.py`, with GeoDistill (DINO) as the base model. Even following the code you provided exactly, I still couldn't reproduce the results shown in Table 1 of the paper. Could you please check the code in the paper? I followed the code you provided for training the Kitti dataset exactly. Initially, shouldn't `teacher` and `student` have no weights?
> 
> The paper explicitly states that the weights for teacher and student are initialized from the same pre-trained base model(the base model is G2SWeakly in this repo, so you should load teacher/student's weight from a pretrained model before retrain with Geodistill), not randomly.

<img width="898" height="382" alt="Image" src="https://github.com/user-attachments/assets/15f54593-61e9-4371-b0c7-6884866fc519" />Sorry, where do the pre-training weights for this teacher and student come from? There is no such pth file
